examples/find_package/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 3.16)
+project(find_package CXX)
+
+if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
```

Comment:
What does this check do? 🤔

Reply:
Ah yes it's super unreadable haha. It's a CMake 3.16-compatible equivalent to [PROJECT_IS_TOP_LEVEL](https://cmake.org/cmake/help/latest/variable/PROJECT_IS_TOP_LEVEL.html). It detects whether or not this project is being included via `add_subdirectory`. If it's included via `add_subdirectory` then I don't want the `find_package` code to be executed since that will fail.

Reply:
I think having to answer this question during review means the code is not obvious so it can be a good idea to add a comment in the file.

Reply:
I agree, since it's not self explanatory, there should be a comment.